snuba_sdk/dsl/dsl.py

```diff
@@ -31,33 +31,34 @@
 coefficient = number / filter
 
 number = ~r"[0-9]+" ("." ~r"[0-9]+")?
-filter = target (open_brace _ condition (_ comma _ condition)* _ close_brace)? (group_by)?
+filter = target (open_brace _ condition (_ comma? _ condition)* _ close_brace)? (group_by)?
 
 condition = condition_op? (variable / tag_key) _ colon _ tag_value
 condition_op = "!"
-tag_key = ~"[a-zA-Z0-9_]+"
-tag_value = quoted_string / quoted_string_tuple / variable
+tag_key = ~r"[a-zA-Z0-9_]+"
+tag_value = quoted_string / unquoted_string / string_tuple / variable
 
-quoted_string = ~r'"([^"\\]*(?:\\.[^"\\]*)*)"'
-quoted_string_tuple = open_square_bracket _ quoted_string (_ comma _ quoted_string)* _ close_square_bracket
+quoted_string = quote unquoted_string quote
```

Comment:
This needs to be changed back to the more open regex above.

The quoted string regex needs to match all special characters and weird stuff, that's the purpose of the quotes.